src/store/index.ts

```diff
@@ -1,6 +1,15 @@
-import { configureStore } from '@reduxjs/toolkit';
-import { reducer } from './reducer.ts';
+import {configureStore} from '@reduxjs/toolkit';
+import {reducer} from './reducer.ts';
+import {createAPI} from '@/services/api.ts';
+
+export const api = createAPI();
 
 export const store = configureStore({
   reducer,
+  middleware: (getDefaultMiddleware) =>
+    getDefaultMiddleware({
+      thunk: {
+        extraArgument: api,
```

Comment:
соответственно при использовании не забыть исправить деструктуризацию

```suggestion
       extraArgument: { api, router }
```